src/main/scala/org/deepdive/ddlog/DeepDiveLogSemanticChecker.scala

```diff
@@ -95,16 +96,39 @@ object DeepDiveLogSemanticChecker extends DeepDiveLogHandler {
     }
     def checkCq(cq: ConjunctiveQuery) {
       checkAtom(cq.head)
-      cq.bodies foreach { x => x foreach checkAtom }
+      checkBodyAtoms(cq.bodies.flatten, checkAtom)
     }
     stmt match {
       case s: ExtractionRule => checkCq(s.q)
       case s: InferenceRule  => checkCq(s.q)
-      case _ => 
+      case _ =>
+    }
+  }
+
+  // check if outer join body contains one atom
+  def checkOuterJoin(stmt: Statement) {
```

Comment:
Can you add a test for this?  Looks like a nice check which deserves one :)  Perhaps after renaming it to something like `checkUselessOptional`?